### PR TITLE
Embed payload repositories as custom repositories

### PR DESCRIPTION
**Pull request by croissanne**
custom repositories are added to /etc/yum.repos.d

---
image

### CODE REVIEW

**Comment by lucasgarfield** [comment]
I just want to make sure I have this correct. 

We use the language 'Custom Repositories' in the Wizard. And for those instances, we are using the camelCase `customRepositories` variable. Variables stored in the form state use kebab-case, which was previously `custom-repositories` and is now `payload-repositories`. This brings it in alignment with the cloud API variable name, `payload_repositories`. 

Given we that the name of the step in the Wizard is "Custom Repositories", it makes sense to keep `customRepositories` and not convert it to `payloadRepositories`.

Just want to double check that this is intentional and not an oversight!

File: src/Components/CreateImageWizard/formComponents/Repositories.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ const BulkSelect = ({
   );
 };
 
-// Utility function to convert from Content Sources to Image Builder API schema
+// Utility function to convert from Content Sources to Image Builder payload repo API schema
 const convertSchemaToImageBuilder = (repo) => {
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/formComponents/Repositories.js:103]: Can we change this to convertSchemaToIBPayloadRepo?

**Comment by lucasgarfield** [comment]
Looks great, everything makes sense! I would just like to request that we rename one function so things are crystal clear for our future selves. :smile:

**Comment by croissanne** [comment]
> I just want to make sure I have this correct.
> 
> We use the language 'Custom Repositories' in the Wizard. And for those instances, we are using the camelCase `customRepositories` variable. Variables stored in the form state use kebab-case, which was previously `custom-repositories` and is now `payload-repositories`. This brings it in alignment with the cloud API variable name, `payload_repositories`.
> 
> Given we that the name of the step in the Wizard is "Custom Repositories", it makes sense to keep `customRepositories` and not convert it to `payloadRepositories`.
> 
> Just want to double check that this is intentional and not an oversight! My first instinct would have been to use sed to do the find replace and it obviously would miss this. smile

exactly, I just wanted to map the variable in the form state to the one we'll send in the api call. But happy to be told otherwise if the naming is less confusing. Plus we can always change it.

File: src/Components/CreateImageWizard/formComponents/Repositories.js (hunk 1 above)
Comment (croissanne) [reply src/Components/CreateImageWizard/formComponents/Repositories.js:103]: absolutely

**Comment by lucasgarfield** [comment]
> exactly, I just wanted to map the variable in the form state to the one we'll send in the api call. But happy to be told otherwise if the naming is less confusing. Plus we can always change it.

I think it is totally fine. Form state variable names match API, UI variables match UI.

**Review by lucasgarfield — APPROVED** [review]
Looks great! Nice work! :)

**Pull request merged**